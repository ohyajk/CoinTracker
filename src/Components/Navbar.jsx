import React from 'react';
import { NavLink } from 'react-router-dom';

const Navbar = () => (
  <div data-testid="Navbar-1" className="flex py-4 px-8 justify-between items-center">
    <NavLink
      to="/"
      className={({ isActive }) => (isActive ? 'invisible' : 'visible')}
    >
      <i className="fa-solid fa-left-long" />
    </NavLink>
    <h2 className="text-2xl md:text-4xl font-black">
      CoinTracker
    </h2>
    <div className="flex justify-center items-center gap-2">
      <i className="fa-solid fa-microphone" />
      <i className="fa-solid fa-gear" />
    </div>
  </div>
);

export default Navbar;
