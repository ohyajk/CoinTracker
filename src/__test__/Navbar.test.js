import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Navbar from '../Components/Navbar';

test('should render navbar', () => {
  const tree = render(
    <BrowserRouter>
      <Navbar />
    </BrowserRouter>,
  );
  expect(tree).toMatchSnapshot();
});
