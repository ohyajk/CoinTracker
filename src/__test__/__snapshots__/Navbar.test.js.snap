// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render navbar 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex py-4 px-8 justify-between items-center"
        data-testid="nav-1"
      >
        <a
          aria-current="page"
          class="invisible"
          href="/"
        >
          <i
            class="fa-solid fa-left-long"
          />
        </a>
        <h2
          class="text-2xl md:text-4xl font-black"
        >
          Coin
          <span
            class="text-violet-400"
          >
            Tracker
          </span>
        </h2>
        <div
          class="flex justify-center items-center gap-2"
        >
          <i
            class="fa-solid fa-microphone"
          />
          <i
            class="fa-solid fa-gear"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex py-4 px-8 justify-between items-center"
      data-testid="nav-1"
    >
      <a
        aria-current="page"
        class="invisible"
        href="/"
      >
        <i
          class="fa-solid fa-left-long"
        />
      </a>
      <h2
        class="text-2xl md:text-4xl font-black"
      >
        Coin
        <span
          class="text-violet-400"
        >
          Tracker
        </span>
      </h2>
      <div
        class="flex justify-center items-center gap-2"
      >
        <i
          class="fa-solid fa-microphone"
        />
        <i
          class="fa-solid fa-gear"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
